main.go

```diff
@@ -54,31 +55,69 @@ func init() {
 }
 
 var flags = struct {
-	EnableLeaderElection bool
-	MetricsAddr          string
+	LeaderElect        bool
+	MetricsBindAddress string
 }{}
 
-func initFlags() {
-	flag.BoolVar(&flags.EnableLeaderElection, "leader-elect", false, "Enable leader election for controller manager.")
-	flag.StringVar(&flags.MetricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
```

Comment:
Unfortunately binding to values directly doesn't work with `viper` so we lose some type safety and flag binding and value binding are separated in the code.